### ISSUE TITLE
Why do I need to add a repo that I don't care about to my software area to get it to build?

### ISSUE THREAD

**Issue by bieryAtFnal** [DUNE-DAQ/daq-cmake#64]
This may be an issue that I've asked about before and can't be fixed, but I thought that I would ask again...

Here are the steps to demonstrate the problem:
```
# start in a fresh shell
# cd to a work area
source /cvmfs/dunedaq.opensciencegrid.org/setup_dunedaq.sh
setup_dbt dunedaq-v2.8.0
dbt-create.sh -n N21-10-12 12OctTest
cd 12OctTest
cd sourcecode
git clone https://github.com/DUNE-DAQ/appfwk.git 
cd appfwk; git checkout feature/NetworkManager; cd ..
git clone https://github.com/DUNE-DAQ/dfmessages.git 
cd dfmessages; git checkout feature/NetworkManager; cd ..
git clone https://github.com/DUNE-DAQ/dfmodules.git 
cd dfmodules; git checkout feature/NetworkManager; cd ..
git clone https://github.com/DUNE-DAQ/networkmanager.git -b develop
git clone https://github.com/DUNE-DAQ/toolbox.git -b develop
git clone https://github.com/DUNE-DAQ/minidaqapp 
cd minidaqapp; git checkout feature/NetworkManager; cd ..
cd ..
sed -i 's/"restcmd" "appfwk"/"restcmd" "toolbox" "ipm" "networkmanager" "appfwk"/' sourcecode/dbt-build-order.cmake
sed -i 's/"ipm" "serialization"/"serialization"/' sourcecode/dbt-build-order.cmake
dbt-workarea-env
dbt-build.sh
```
This gives an error like the following:
```
CMake Error at networkmanager/CMakeLists.txt:8 (find_package):
  Found package configuration file:

    /cvmfs/dunedaq-development.opensciencegrid.org/nightly/N21-10-12/packages/ipm/08e93c1/slf7.x86_64.e19.prof/ipm/lib64/ipm/cmake/ipmConfig.cmake

  but it set ipm_FOUND to FALSE so package "ipm" is considered to be NOT
  FOUND.  Reason given by package:

  The following imported targets are referenced, but are missing:
  logging::logging

-- Configuring incomplete, errors occurred!

```
If I add the IPM repo, that helps
```
cd sourcecode
git clone https://github.com/DUNE-DAQ/ipm.git -b develop 
cd ..
dbt-build.sh
```

**Comment by jcfreeman2** [comment]
tl;dr This was fixed a couple of months ago. Developers weren't adding `find_dependency` calls to their `<package>Config.cmake.in` files which were required. 

The problem is that to build a repo in a workarea, it not only needs to know where its immediate dependencies are, but also the dependencies of its dependency. As described in https://dune-daq-sw.readthedocs.io/en/latest/packages/daq-cmake/#installing-your-project-as-a-local-package, if you're writing a package someone else will use (i.e., they'll add `find_package(<your package>)` to their `CMakeLists.txt`) then if your package itself has dependencies, you need to add them to `<package>Config.cmake.in` via `find_dependency` calls. So if you've written package `Beta` which depends on package `Gamma`, and someone else wants to write a package `Alpha` which depends on package `Beta`, then when their `CMakeLists.txt` file calls
```
find_package(Alpha)
```
it'll get its hands on `BetaConfig.cmake`. But if `Beta` depends on `Gamma` then `Alpha` will need to know how to pull in `Gamma`'s objects, and that's done by you adding 
```
find_dependency(Gamma)
```
to `BetaConfig.cmake.in`
